# Document Report
** Created by - Sanjana Pal
- date - 02 may
- code realted to -Assignment topic - Hypothesis testing.

# Hypothesis Testing

### Q1      A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.


## inference 
### Step 1 - Formulating the null and alternate hypothesis
- Assume null hypothesis is Ho = μ1 = μ2( there is no difference in the diameter of the cutlets between two units.)
 -  Alternate Hypothesis Ha = μ1 ≠ μ2 (There is significance difference in the diameter of the cutles between two units.)
- IT is a two tail test

### given 

- CI = 95%,so that significance level is 5% = 0.05


In [4]:
import pandas as pd
import numpy as np
from scipy import stats

In [ ]:
df = pd.read_csv("Cutlets.csv")

In [ ]:
df

In [ ]:
len(df)

In [ ]:
### Test Statistics for given hypothesis
### Step 2
- There are two independent samples unit A and unit B 
- Population deviation is not given
- Sample Size is small
- Considering all of this we will using two sample t test with one tail

In [ ]:
# 2 sample 2 tail_ttest:   stats.ttest_ind(array1,array2)  #ind = independent sample

p_value=stats.ttest_ind(df['Unit A'].squeeze(),df['Unit B'].squeeze()) # squeeze is used to convert column into series

In [ ]:
p_value

In [ ]:
pvalue = 0.47 and alpha value is 0.05

 - Above equation shows that pvalue is greater than alpha value,so we conclude that we are failed to reject null hypothesis
    ,its shows there is no differnce in the cutlets between two units.

## Q2.A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
 

In [ ]:
dftat = pd.read_csv("LabTAT.csv")

In [ ]:
dftat

In [ ]:
### Step 1 - Formulating null and alternate hypothesis
- Null hypothesis = m1=m2=m3=m4
- Alternate hypothesis = m1 =m2= m3 !=m4
Anova ftest = Analysis of more than 2 samples or columns.

In [ ]:
p_values = stats.f_oneway(dftat.iloc[:,0],dftat.iloc[:,1],dftat.iloc[:,2],dftat.iloc[:,3])

In [ ]:
p_values

- comparing between p value and significance difference i.e p>alpha(2.11>0.05),pvalue is greater than alpha ,its shows we failed to reject Null hypothesis ,so we can say that average TAT of reports of the laboratories is the same.

## Q.3  Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

In [5]:
dfby = pd.read_csv("BuyerRatio (1).csv")

In [6]:
dfby

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

## Formulating Null and Alternate Hypothesis
- Null hypothesis Ho = male and female buyer ratio is same 
- Alternate hypothesis = male and female buyer ratio is different.

In [12]:
dfby['Observed Values'] = dfby['Observed Values'].astype('category')
dfby = dfby.set_index('Observed Values')# note for chisquare it takes all int values so always our category should be set as index 

In [13]:
from scipy.stats import chi2_contingency

In [16]:
chi2_stat, p_val ,dof , expected = chi2_contingency(dfby)

In [17]:
print(f'chi-squared statistics:{chi2_stat}')
print(f'p_value:{p_val}')
print(f'degree of freedom :{dof}')
print(f' expected frequency:{expected}')

chi-squared statistics:1.595945538661058
p_value:0.6603094907091882
degree of freedom :3
 expected frequency:[[  42.76531299  146.81287862  131.11756787   72.30424052]
 [ 442.23468701 1518.18712138 1355.88243213  747.69575948]]


In [ ]:
p_valu = 0.66

# comparing p_value and alpha value
- p_valu=0.66 and alpha value =0.05 ,so basically p_value is greater than alpha ,we failed to reject null hypothesis.
In other word we don't have sufficient evidence to support the alternative hypothesis and we conclude that there is no significant difference between observed value and expected value all the proportion are equal

- In the context of the chi-squared test, this means there is no differnce between observe and expected fequencies.Therefore we conclude that there is no evidence pf relationship between variable being studied.

# Q.4 TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


- Null hypothesis Ho = customer order form defective% does not varies by center
- Alternate hypothesis Ha = customer order form defevtive% varies by center.
- Significance level = 0.05

In [22]:
data = pd.read_csv("Costomer+OrderForm (1).csv")

In [23]:
data

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [25]:
data['Phillippines'].value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [26]:
data['Indonesia'].value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [27]:
data['Malta'].value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [28]:
data['India'].value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [31]:
obs = np.array([[271,267,269,280],[29,33,31,20]])

In [32]:
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [33]:
chi2_stats, p_value, dof, expected = chi2_contingency(obs)

In [34]:
chi2_stats

3.858960685820355

In [35]:
p_value

0.2771020991233135

In [36]:
dof

3

In [37]:
expected

array([[271.75, 271.75, 271.75, 271.75],
       [ 28.25,  28.25,  28.25,  28.25]])

## comparing p_value and alpha value

- p_ value = 0.277,
- alpha value = 0.05,
p_value > alpha value, we conclude that p value is greater than alpha value ,so we fail to reject null hypothesis i.e customer order form defective % does not varies by center. 